core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -824,6 +809,9 @@ public boolean load(String graphHopperFolder) {
     }
 
     private void checkProfilesConsistency() {
+        if (profilesByName.isEmpty()) {
```

Comment:
There are a few odd cases where we do not use `GraphHopper` for routing (like `NearestResource`) so these do not necessarily require profiles. Not sure, to me this seems rather the exception so I am checking the existence of at least one profile here. Once we do #1901 we can move this check into `GraphHopperRouter`'s constructor for example.